### ISSUE TITLE
AccessibleName not working for RichEditBox

### ISSUE THREAD

**Issue by ayushanand-microsoft** [dotnet/winforms#9918]
### .NET version

.NET - 7.0.400
.NET Framework - 4.7.2

### Did it work in .NET Framework?

No

### Issue description

AccessibleName property is not working in RichEditBox control.

### Steps to reproduce

1. Create a Form with a RichEditBox.
2. Provide a name in the AccessibleName property.
3. Run the app.

**Actual behavior:**
RichEditBox doesn't reflect the AccessibleName and UIA always shows the name as RichEdit Control.

![image](https://github.com/dotnet/winforms/assets/144995455/a77d1fd2-c52b-4b6f-9bdd-508a52966d65)

**Expected behavior:**
The name of the RichEditBox control should be the one specified as AccesibleName.

**Comment by elachlan** [comment]
> RichTextBox: at this moment, we don't provide a managed accessible object for it. It doesn't support setting a custom AccessibleRole and changing "ControlType" respectively (in .NET Framework 4.7.2 too). It always has "Role" as Document and "ControlType" as Document.

https://github.com/dotnet/winforms/blob/ac5df254ecadb04a1c89c14e23340c95160ba13f/docs/accessibility/accessible-role-controltype.md#exceptional-cases-and-additional-info

https://github.com/dotnet/winforms/blob/ac5df254ecadb04a1c89c14e23340c95160ba13f/src/System.Windows.Forms/src/System/Windows/Forms/RichTextBox.cs#L2434-L2435

I looked into it a bit. This is what I found.

**Comment by merriemcgaw** [comment]
@ayushanand-microsoft please note that this is something we'd be able to fix for .NET 9 because we're too close to the end of .NET 8. 

Is the application in the image one that is built on .NET 7 or is it built on .NET Framework?

FYI @Tanya-Solyanik and @LeafShi1

**Comment by Tanya-Solyanik** [comment]
@LeafShi1  - please investigate if there is a workaround and share your findings.

**Comment by Tanya-Solyanik** [comment]
@LeafShi1  - please investigate if it's possible to set AccessibleName by PInvoking native methods?

**Comment by ayushanand-microsoft** [comment]
Hi Merrie,

Our app is built on .NET Framework.

Thanks,
Ayush

**Comment by LeafShi1** [comment]
> @LeafShi1 - please investigate if it's possible to set AccessibleName by PInvoking native methods? We can't implement a UIA Provider that would be working as well as the native one because we don't have access to the internal properties of the RichTextBox. So providing a workaround is the only reasonable solution here.

I searched for oleacc.dll and shobjidl.dll, which contain methods for setting AccessibleName. However, according to the [official documentation](https://learn.microsoft.com/en-us/windows/win32/api/oleacc/nf-oleacc-iaccessible-put_accname), the method "put_accName" in oleacc.dll is no longer supported. shobjidl.dll cannot be found in operating system.

**Comment by Tanya-Solyanik** [comment]
@ayushanand-microsoft  - Looks like the only way to provide a custom accessible name in a text label right before this RichTextBox in the tab order.
